src/shape_inference.jl

```diff
@@ -379,6 +379,27 @@ for func in ["Sum", "Prod", "Min", "Max", "All", "Any", "Mean"]
     end
 end
 
+for func in ["ArgMax", "ArgMin"]
+    register_shape(func) do op
+        value_shape = copy(_get_shape(get_input(op, 1)))
+        reduction_dims = get_input(op, 2)
+        if value_shape.rank_unknown
+            return [TensorShape(nothing)]
+        end
+        reduction_dim_values = load_const(reduction_dims)
+        if isnull(reduction_dim_values)
```

Comment:
I think cleaner if you worte:

```
if value_shape.rank_unknown || isnull(reduction_dim_values)
       return [TensorShape(nothing)]
end
```

and removed the earlier if

Reply:
agreed